client/one-time-payment/react/src/context/sdkContext.tsx

```diff
@@ -33,29 +61,13 @@ export const PayPalSDKProvider: React.FC<PayPalSDKProviderProps> = ({
   const [eligiblePaymentMethods, setEligiblePaymentMethods] =
     useState<EligiblePaymentMethods | null>(null);
 
-  const initSdkInstance = async () => {
-    const sdkInstance: SdkInstance = await window.paypal.createInstance({
-      clientToken,
-      components,
-      pageType,
-    });
-
-    const eligiblePaymentMethods = await sdkInstance.findEligibleMethods({
-      currency: "USD",
-    });
-
-    return {
-      sdkInstance,
-      eligiblePaymentMethods,
-    };
-  };
-
   useEffect(() => {
     const loadPayPalSDK = async () => {
       if (!sdkInstance && clientToken) {
         try {
-          const { sdkInstance, eligiblePaymentMethods } =
-            await initSdkInstance();
+          const { sdkInstance, eligiblePaymentMethods } = await initSdkInstance(
+            { clientToken, components, pageType },
```

Comment:
One slight concern I have about this is that `components ` and `pageType ` aren't in the dependency array for this `useEffect`, so if they change the instance won't change. But maybe that's ok?

Reply:
Thats a great call out. I do think a new sdkInstance should be created when `components` or `pageType` change. I added a commit to add these two params to the dependency array.